libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -258,14 +286,22 @@ void ImmutableValidator::checkAllVariablesInitialized(solidity::langutil::Source
 	{
 		for (VariableDeclaration const* varDecl: contract->stateVariables())
 			if (varDecl->immutable())
-				if (!util::contains(m_initializedStateVariables, varDecl))
+			{
+				if (util::contains(m_stateVariablesNotInitializedInBothBranches, varDecl))
+					m_errorReporter.typeError(
+						4599_error,
+						varDecl->nameLocation(),
+						"Immutable is not initialized in every branch of control flow."
+					);
+				else if (util::contains(m_stateVariablesNotInitializedInBothBranches, varDecl) ||
```

Comment:
Removed `util::contains(m_stateVariablesNotInitializedInBothBranches, varDecl)` -- it's redundant